protocol/src/main/java/org/geysermc/mcprotocollib/network/tcp/TcpPacketCodec.java

```diff
@@ -23,57 +30,87 @@ public TcpPacketCodec(Session session, boolean client) {
 
     @SuppressWarnings({"rawtypes", "unchecked"})
     @Override
-    public void encode(ChannelHandlerContext ctx, Packet packet, ByteBuf buf) {
-        int initial = buf.writerIndex();
+    public void encode(ChannelHandlerContext ctx, Packet packet, List<Object> out) {
+        if (log.isTraceEnabled()) {
+            log.trace("Encoding packet: {}", packet.getClass().getSimpleName());
+        }
```

Comment:
do we need the trace enabled check? doesn't log.trace() perform it too before formatting the string?

Reply:
This is to avoid array allocations and expensive method calls, packet.getClass().getSimpleName() will lag a tiny bit and the trace enabled check prevents it being called.